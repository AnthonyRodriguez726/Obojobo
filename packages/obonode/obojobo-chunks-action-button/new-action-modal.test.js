import React from 'react'
import { mount } from 'enzyme'

import AddActionModal from './new-action-modal'

describe('ActionButton Properties Modal', () => {
	beforeEach(() => {
		jest.clearAllMocks()
	})

	test('AddActionModal component', () => {
		const component = mount(<AddActionModal />)
		const tree = component.html()

		expect(tree).toMatchSnapshot()
	})

	test('AddActionModal component checks the value before confirming', () => {
		const onConfirm = jest.fn()

		const component = mount(<AddActionModal onConfirm={onConfirm} />)

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).not.toHaveBeenCalled()
	})

	test('AddActionModal component changes the value and confirms', () => {
		const onConfirm = jest.fn()

		const component = mount(<AddActionModal onConfirm={onConfirm} />)

		component
			.find('textarea')
			.at(0)
			.simulate('change', { target: { value: '{"id":"mockId"}' } })

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).toHaveBeenCalled()
	})

	test('AddActionModal component changes the type to nav:prev', () => {
		const onConfirm = jest.fn()

		const component = mount(<AddActionModal onConfirm={onConfirm} />)

		component
			.find('select')
			.at(0)
			.simulate('change', { target: { value: 'nav:prev' } })

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).toHaveBeenCalled()
	})

	test('AddActionModal component changes the type to nav:openExternalLink and fails to verify', () => {
		const onConfirm = jest.fn()

		const component = mount(<AddActionModal onConfirm={onConfirm} />)

		component
			.find('select')
			.at(0)
			.simulate('change', { target: { value: 'nav:openExternalLink' } })

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).not.toHaveBeenCalled()
	})

	test('AddActionModal component changes the type to nav:openExternalLink and verifys', () => {
		const onConfirm = jest.fn()

		const component = mount(<AddActionModal onConfirm={onConfirm} />)

		component
			.find('select')
			.at(0)
			.simulate('change', { target: { value: 'nav:openExternalLink' } })

		component
			.find('textarea')
			.at(0)
			.simulate('change', { target: { value: '{"url":"mockURL"}' } })

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).toHaveBeenCalled()
	})

	test('AddActionModal component changes the type to assessment:startAttempt and fails to verify', () => {
		const onConfirm = jest.fn()

		const component = mount(<AddActionModal onConfirm={onConfirm} />)

		component
			.find('select')
			.at(0)
			.simulate('change', { target: { value: 'assessment:startAttempt' } })

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).not.toHaveBeenCalled()
	})

	test('AddActionModal component changes the type to assessment:startAttempt and verifys', () => {
		const onConfirm = jest.fn()

		const component = mount(<AddActionModal onConfirm={onConfirm} />)

		component
			.find('select')
			.at(0)
			.simulate('change', { target: { value: 'assessment:startAttempt' } })

		component
			.find('textarea')
			.at(0)
			.simulate('change', { target: { value: '{"id":"mockID"}' } })

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).toHaveBeenCalled()
	})

	test('AddActionModal component changes the type to assessment:endAttempt and fails to verify', () => {
		const onConfirm = jest.fn()

		const component = mount(<AddActionModal onConfirm={onConfirm} />)

		component
			.find('select')
			.at(0)
			.simulate('change', { target: { value: 'assessment:endAttempt' } })

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).not.toHaveBeenCalled()
	})

	test('AddActionModal component changes the type to assessment:endAttempt and verifys', () => {
		const onConfirm = jest.fn()

		const component = mount(<AddActionModal onConfirm={onConfirm} />)

		component
			.find('select')
			.at(0)
			.simulate('change', { target: { value: 'assessment:endAttempt' } })

		component
			.find('textarea')
			.at(0)
			.simulate('change', { target: { value: '{"id":"mockID"}' } })

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).toHaveBeenCalled()
	})

	test('AddActionModal component changes the type to viewer:alert and fails to verify', () => {
		const onConfirm = jest.fn()

		const component = mount(<AddActionModal onConfirm={onConfirm} />)

		component
			.find('select')
			.at(0)
			.simulate('change', { target: { value: 'viewer:alert' } })

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).not.toHaveBeenCalled()
	})

	test('AddActionModal component changes the type to viewer:alert and verifys', () => {
		const onConfirm = jest.fn()

		const component = mount(<AddActionModal onConfirm={onConfirm} />)

		component
			.find('select')
			.at(0)
			.simulate('change', { target: { value: 'viewer:alert' } })

		component
			.find('textarea')
			.at(0)
			.simulate('change', { target: { value: '{"title":"mockTitle", "message":"mockMessage"}' } })

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).toHaveBeenCalled()
	})

	test('AddActionModal component focuses on first element', () => {
		const component = mount(<AddActionModal content={{}} />)

		component.instance().focusOnFirstElement()

		expect(component.html()).toMatchSnapshot()
	})
})
