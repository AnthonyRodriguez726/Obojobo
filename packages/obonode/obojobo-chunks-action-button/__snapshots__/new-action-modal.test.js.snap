// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton Properties Modal AddActionModal component 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog\\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">New Action</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"new-action\\"><div class=\\"trigger-name\\"><label for=\\"obojobo-draft--chunks--action-button--action\\">Action:</label><select id=\\"obojobo-draft--chunks--action-button--action\\" name=\\"Action\\"><option value=\\"nav:goto\\">nav:goto</option><option value=\\"nav:prev\\">nav:prev</option><option value=\\"nav:next\\">nav:next</option><option value=\\"nav:openExternalLink\\">nav:openExternalLink</option><option value=\\"nav:lock\\">nav:lock</option><option value=\\"nav:unlock\\">nav:unlock</option><option value=\\"nav:open\\">nav:open</option><option value=\\"nav:close\\">nav:close</option><option value=\\"nav:toggle\\">nav:toggle</option><option value=\\"assessment:startAttempt\\">assessment:startAttempt</option><option value=\\"assessment:endAttempt\\">assessment:endAttempt</option><option value=\\"viewer:alert\\">viewer:alert</option></select></div><div class=\\"trigger-value\\"><label for=\\"obojobo-draft--chunks--action-button--value\\">Value:</label><textarea name=\\"Value\\" id=\\"obojobo-draft--chunks--action-button--value\\" placeholder=\\"{&quot;id&quot;:&quot;page-id&quot;}\\"></textarea></div><span></span></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`ActionButton Properties Modal AddActionModal component focuses on first element 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog\\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">New Action</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"new-action\\"><div class=\\"trigger-name\\"><label for=\\"obojobo-draft--chunks--action-button--action\\">Action:</label><select id=\\"obojobo-draft--chunks--action-button--action\\" name=\\"Action\\"><option value=\\"nav:goto\\">nav:goto</option><option value=\\"nav:prev\\">nav:prev</option><option value=\\"nav:next\\">nav:next</option><option value=\\"nav:openExternalLink\\">nav:openExternalLink</option><option value=\\"nav:lock\\">nav:lock</option><option value=\\"nav:unlock\\">nav:unlock</option><option value=\\"nav:open\\">nav:open</option><option value=\\"nav:close\\">nav:close</option><option value=\\"nav:toggle\\">nav:toggle</option><option value=\\"assessment:startAttempt\\">assessment:startAttempt</option><option value=\\"assessment:endAttempt\\">assessment:endAttempt</option><option value=\\"viewer:alert\\">viewer:alert</option></select></div><div class=\\"trigger-value\\"><label for=\\"obojobo-draft--chunks--action-button--value\\">Value:</label><textarea name=\\"Value\\" id=\\"obojobo-draft--chunks--action-button--value\\" placeholder=\\"{&quot;id&quot;:&quot;page-id&quot;}\\"></textarea></div><span></span></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;
